### DISCUSSION TITLE
Display Roles in a more neater way

### DISCUSSION THREAD

**Discussion by samjesus8** [DSharpPlus/DSharpPlus#1475] [Support] [answered]
I currently have this profile command which displays all the normal information about your account but I'm trying to find a way to make the role section look a bit neater. Is there a way I can combine both the role name & color and display it in a nice manner

# Command Code
```
[Command("profile")]
        public async Task ProfileCommand(CommandContext ctx) 
        {
            //Fields
            var UserName = ctx.User.Username;
            var JoinedAtDate = ctx.Member.JoinedAt.Date.ToShortDateString();
            var CreatedDate = ctx.User.CreationTimestamp.Date.ToShortDateString();
            var Roles = ctx.Member.Roles;

            //Conversions
            var RoleConvert = Roles.ToArray();
            string[] roleList = new string[RoleConvert.Length];
            for(int i = 0; i < roleList.Length; i++) 
            {
                roleList[i] = RoleConvert[i].Name;
            }

            //Message
            var profileMessage = new DiscordMessageBuilder()
                .AddEmbed(new DiscordEmbedBuilder()
                .WithColor(DiscordColor.Azure)
                .AddField("Username", UserName, true)
                .AddField("Joined Guild", JoinedAtDate, true)
                .AddField("Account Created", CreatedDate, true)
                .AddField("Level", "COMING SOON", true)
                .AddField("Balance", "COMING SOON", true)
                .AddField("Roles", string.Join(", ", roleList), false)
                .WithThumbnail(ctx.User.AvatarUrl)
                );

            await ctx.Channel.SendMessageAsync(profileMessage);
        }
```

# Output

The current look of the message (in a public server)
![image](https://user-images.githubusercontent.com/98812930/221031499-eaf35cdb-9a17-47f6-8232-a44a2e9e1cf5.png)

**Comment by OoLunar** [comment, marked as answer]
You can ping the role through `DiscordRole.Mention`. By default, pings that are in embeds will not actually ping the associated people, only display it. If you wish to be double sure, you could use a `DiscordMessageBuilder` with `AllowedMentions(Mentions.None)`.

**Comment by OoLunar** [comment]
After further inspection, I can't help but notice one of your role names. Could you provide an explanation on that please?

**Reply by samjesus8** [reply]
> After further inspection, I can't help but notice one of your role names. Could you provide an explanation on that please?

Oh dont worry about that. We have a genshin chat in one of the servers I'm in and they hate the game so much they made a role called pedophiles and anyone who had it had access to the chat. Its nothing that deep lol

**Reply by OoLunar** [reply]
Please refrain from sharing such content within future discussions as it can impact the type of support you receive and how reluctant others may be to interact with you.

**Reply by samjesus8** [reply]
Right no worries. It was my mistake to even use it as an example but it was on the fly. Anyways like I said it's not meant to be taken seriously
